In [1]:
import numpy as np
import pandas as pd

##  Assign it to a variable called users and use the 'user_id' as index 

In [2]:
users = pd.read_csv("https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user")
users.head()

user_id|age|gender|occupation|zip_code
0                1|24|M|technician|85711
1                     2|53|F|other|94043
2                    3|23|M|writer|32067
3                4|24|M|technician|43537
4                     5|33|F|other|15213

In [3]:
columns = []
for i in users.columns:
    columns.append(i.split("|"))
print(columns)

[['user_id', 'age', 'gender', 'occupation', 'zip_code']]


In [4]:
values = []
for c in users["user_id|age|gender|occupation|zip_code"].values:
    values.append(c.split("|"))


In [5]:
new_column = columns[0]

In [6]:
user_id = []
age = []
gender = []
occupation = []
zip_code = []

for i in range(0, len(values)):
    user_id.append(values[i][0])
    age.append(values[i][1])
    gender.append(values[i][2])
    occupation.append(values[i][3])
    zip_code.append(values[i][4])


In [7]:
users = pd.DataFrame({ new_column[1]: age, new_column[2]: gender, new_column[3]: occupation, 
                      new_column[4]: zip_code}, index = user_id)

In [8]:
users.head()

age gender  occupation zip_code
1  24      M  technician    85711
2  53      F       other    94043
3  23      M      writer    32067
4  24      M  technician    43537
5  33      F       other    15213

## See the first 25 entries 

In [9]:
users.head(25)

age gender     occupation zip_code
1   24      M     technician    85711
2   53      F          other    94043
3   23      M         writer    32067
4   24      M     technician    43537
5   33      F          other    15213
6   42      M      executive    98101
7   57      M  administrator    91344
8   36      M  administrator    05201
9   29      M        student    01002
10  53      M         lawyer    90703
11  39      F          other    30329
12  28      F          other    06405
13  47      M       educator    29206
14  45      M      scientist    55106
15  49      F       educator    97301
16  21      M  entertainment    10309
17  30      M     programmer    06355
18  35      F          other    37212
19  40      M      librarian    02138
20  42      F      homemaker    95660
21  26      M         writer    30068
22  25      M         writer    40206
23  30      F         artist    48197
24  21      F         artist    94533
25  39      M       engineer    55107

## See the last 10 entries

In [10]:
users.tail(10)

age gender     occupation zip_code
934  61      M       engineer    22902
935  42      M         doctor    66221
936  24      M          other    32789
937  48      M       educator    98072
938  38      F     technician    55038
939  26      F        student    33319
940  32      M  administrator    02215
941  20      M        student    97229
942  48      F      librarian    78209
943  22      M        student    77841

##  What is the number of observations in the dataset? 

In [11]:
users.shape

(943, 4)

## What is the number of columns in the dataset? Print the name of all the columns.

In [13]:
users.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

## How is the dataset indexed? 

using user_id the data is indexed

## What is the data type of each column? 

In [15]:
users.columns.dtype

dtype('O')

## Print only the occupation column 

In [16]:
users["occupation"]

1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

## How many different occupations are in this dataset? 

In [19]:
users['occupation'].describe(include='all')

count         943
unique         21
top       student
freq          196
Name: occupation, dtype: object

There are 21 different occupations in this data set

## What is the most frequent occupation? 

Student with a frequency of 196 times

## Summarise the Data 

In [20]:
users.describe()

age gender occupation zip_code
count   943    943        943      943
unique   61      2         21      795
top      30      M    student    55414
freq     39    670        196        9

## Mean Age of Users 

In [37]:
convert_dict = {"age": int}
users = users.astype(convert_dict)

In [39]:
users.describe()

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000